### PR TITLE
WT-13557 Move evict_inline.h and conn_evict.c in src/evict

### PR DESCRIPTION
**Pull request by Siddhartha8899**
Move src/incude/evict_inline.h and src/conn/conn_evict.c into src/evict/

Starting commit for this PR - [Move evict_inline.h and evict_conn.c to src/evict/](https://github.com/wiredtiger/wiredtiger/pull/11044/commits/d938b2c0407c684638e76e2ea5b41f35f3fc35a8)

This change moves eviction logic that was defined in src/conn/conn_evict.c and src/include/evict_inline.h into the eviction module for consistency.
As a result:

The following functions are now private to eviction and not part of the public API

```
__wt_btree_dominating_cache     -> __wti_btree_dominating_cache
__wt_cache_eviction_worker      -> __wti_cache_eviction_worker
__wt_cache_hs_dirty             -> __wti_cache_hs_dirty
__wt_cache_read_gen             -> __cache_read_gen
__wt_cache_read_gen_incr        -> __wti_cache_read_gen_incr
__wt_eviction_dirty_target      -> __wti_eviction_dirty_target
__wt_eviction_updates_needed    -> __wti_eviction_updates_needed
```
The following functions have moved out of the connection module and are part of the public evict API
```
__wti_evict_config                 -> __wt_evict_config
__wti_evict_create                 -> __wt_evict_create
__wti_evict_destroy               -> __wt_evict_destroy
__wti_evict_stats_update      -> __wt_evict_stats_update
```
Finally two function remains part of the public API but has been renamed for clarity.
```
__wti_evict_create               -> __wt_evict_threads_create
__wti_evict_destroy             -> __wt_evict_threads_destroy
```

### CODE REVIEW

**Comment by Siddhartha8899** [comment]
Starts from - [d938b2c](https://github.com/wiredtiger/wiredtiger/pull/11044/commits/d938b2c0407c684638e76e2ea5b41f35f3fc35a8)

File: src/evict/evict_inline.h [hunk 1]
```diff
@@ -234,11 +233,11 @@ __wt_eviction_updates_needed(WT_SESSION_IMPL *session, double *pct_fullp)
 }
 
 /*
- * __wt_btree_dominating_cache --
+ * __wti_btree_dominating_cache --
```
Comment (ajmorton) [inline src/evict/evict_inline.h:236]: `__wti_btree_dominating_cache` is only used once in evict_lru.c and that's unlikely to change given the specificity of the function. If we move it to evict_lru.c that's one less function defined in exposed in extern.h.

Happy to push this out to a subsequent PR and keep the current PR scope small.

**Review by ajmorton — APPROVED** [review]
LGTM 👍 
Could you add the following text to the PR description?

"""
This change moves eviction logic that was defined in `src/conn/conn_evict.c` and `src/include/evict_inline.h` into the eviction module for consistency. 
As a result:

The following functions are now private to eviction and not part of the public API
```
__wt_btree_dominating_cache     -> __wti_btree_dominating_cache
__wt_cache_eviction_worker      -> __wti_cache_eviction_worker
__wt_cache_hs_dirty             -> __wti_cache_hs_dirty
__wt_cache_read_gen             -> __cache_read_gen
__wt_cache_read_gen_incr        -> __wti_cache_read_gen_incr
__wt_eviction_dirty_target      -> __wti_eviction_dirty_target
__wt_eviction_updates_needed    -> __wti_eviction_updates_needed
```

The following functions have moved out of the connection module and are part of the public evict API
```
__wti_evict_config              -> __wt_evict_config
__wti_evict_create              -> __wt_evict_create
__wti_evict_destroy             -> __wt_evict_threads_destroy
__wti_evict_stats_update        -> __wt_evict_stats_update
```

Finally one function remains part of the public API but has been renamed for clarity.
```
__wti_evict_destroy             -> __wt_evict_threads_destroy
```

File: dist/dist.py [hunk 2]
```diff
@@ -7,7 +7,7 @@
 #    Return a list of the WiredTiger source file names.
 def source_files():
     file_re = re.compile(r'^\w')
-    for line in glob.iglob('../src/include/*.h'):
+    for line in glob.iglob('../src/*/*.h'):
```
[the remark this answers is not in the document]
Comment (Siddhartha8899) [reply dist/dist.py:10]: Yes, nice catch, for functions we do skip *checksum* directory, but not utilities, however, none of the utilities functions start with __wt_ or __wti.

File: dist/filelist [hunk 3]
```diff
@@ -105,6 +104,7 @@ src/cursor/cur_stat.c
 src/cursor/cur_std.c
 src/cursor/cur_table.c
 src/cursor/cur_version.c
+src/evict/evict_conn.c
```
[the remark this answers is not in the document]
Comment (Siddhartha8899) [reply dist/filelist:107]: `evict_core`? It's hard to come up with a name for this one.

File: src/evict/evict_inline.h [hunk 4]
```diff
@@ -20,21 +20,21 @@ __wt_cache_aggressive(WT_SESSION_IMPL *session)
 }
 
 /*
- * __wt_cache_read_gen --
+ * __cache_read_gen --
```
[the remark this answers is not in the document]
Comment (Siddhartha8899) [reply src/evict/evict_inline.h:23]: Not as part of this PR, it's part of the next PR.

File: src/evict/evict_inline.h (hunk 1 above)
Comment (Siddhartha8899) [reply src/evict/evict_inline.h:236]: Good catch, I will leave this to the next PR as well.

File: dist/filelist (hunk 3 above)
[the remark this answers is not in the document]
Comment (ajmorton) [reply dist/filelist:107]: Agreed. I'm happy to stick with `evict_conn.c` unless someone comes along with something good. We can merge the PR without resolving this thread.

**Review by ershov — APPROVED** [review]
LGTM!
Although I keep getting lost in what part of the overall change a particular PR is.

**Pull request merged**